project_template/BUILDING.md

```diff
@@ -0,0 +1,12 @@
+
+## Requirements
+
+This azure sdk module has several external build-time requirements.
+
+- **Catch2** (for tests): https://github.com/catchorg/Catch2.git

```

Comment:
If the desire is to standardize on a single testing framework, Catch is less likely to be sufficient to cover all requirements (for example, death tests). Instead, I would recommend Google Test as the de-facto mature, complete, and maintained testing library. Additionally, Visual studio has explicit support for google test and does not have that support for catch.

If there isn't a desire to standardize on a single testing framework, then suggesting Catch2 here is less problematic.

Reply:
yeah, I can switch it to gtest, but ultimately it's similar.

Reply:
If we decide to standardize, then we need to have a deeper discussion. Our test framework should also be portable and small enough to run on some of our devices. Catch2 itself doesn't seem to be the best choice for a C library, since it does not support C officially. We have a homegrown test and mock frameworks which we have been using for 4+ years which could also be an option.

---
In reply to: [308971910](https://github.com/Azure/azure-sdk-for-c/pull/2#discussion_r308971910) [](ancestors = 308971910)